docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,190 @@
+any configuration option you have set, this command will also perform a series of optimizations to deliver the best runtime when starting and running the server. Basically,
+a lot of processing that would usually happen when starting/running the server is no longer necessary and the server can start and run faster.
+
+Some optimizations performed by the `build` command are:
+
+* Closed-world assumption about installed providers, hence no need to re-create the registry everytime the server starts
+* Configuration files (e.g.: clustering) are pre-parsed to reduce IO when starting the server
+* Database specific resources are configured and prepared to run against a specific database vendor
+* By persisting configuration options into the server image, the server does not perform any additional step to interpret configuration options and (re)configure itself
+
+There are many more optimizations, most of them performed by <@link.external url="https://quarkus.io" text="Quarkus"/> itself. Just keep in mind that this additional step is a key aspect to achieve the optimal runtime,
+ideally when running Keycloak as a container.
+
+Once you run the `build` command, you won't need to set the same configuration options again (they are now persisted) so that you can just `start` the server:
+
+.Building an optimized server image
+<@kc.build parameters="--db=postgres --db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+.Starting the server
+<@kc.start/>
+
+As a rule of thumb, prefer building a server image to achieve the optimal runtime as well as running the `start` command without any parameter for faster startup time.
+
+=== Overriding settings
+
+Once you have built a server image using the `build` command, you should be able to override any property that is available from the `start` command.
+
+For instance, if you have built an image as follows:
+
+<@kc.build parameters="--db=postgres --db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+You should be able to override the `db-username`, `hostname`, or any other property available to the `start` command:
+
+<@kc.start parameters="--db-username foo --hostname newkeycloak.acme.com"/>
+
+As a result, the server should start using all the configuration options previously set using the `build` command but overriding those that you explicitly
+set when starting the server.
+
+This configuration approach can be useful when building an opinionated server image with the possibility to override specific options that are allowed to
+change when starting the server (those available to the `start` command).
+
+== Using placeholders
+
+You should be able to use placeholders to dynamically change the value of configuration options when starting the server.
+
+For that, you should use the following format when defining the value of an option:
+
+```
+db.username=${r"${my.db.user}"}
```

Comment:
When I use this syntax, I get 
```
ERROR: Build failure: Build failed due to errors
	[error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#persistBuildTimeProperties threw an exception: java.util.NoSuchElementException: SRCFG00011: Could not expand value r"mydbuser" in property kc.db.username
```

Typo? Should be ${my.db.user}, shouldn't it? Also: I'd prefer the example to be with an environment variable instead of a system property. So ${MY_DB_USER}. Imo we shouldn't encourage the usage of system properties, but that's just my opinion.

Reply:
If you try to render the doc you should get what you want. Could you please check?

W.r.t. using env vs props, in general I agree. But here we are explaining that placeholders can be resolved from both sys props and env vars.

Otherwise we would be limiting to only env vars?